wallet/src/de/schildbach/wallet/ExploreSyncWorker.kt

```diff
@@ -106,44 +117,53 @@ class ExploreSyncWorker constructor(val appContext: Context, workerParams: Worke
             val merchantDataSize = exploreRepository.getMerchantDataSize()
-            if (merchantDataSizeDB == 0) {
-                merchantDao.deleteAll()
-            } else {
-                log.info("MERCHANT data partially synced ($merchantDataSizeDB of $merchantDataSize), continuing")
-            }
-            val merchantCache = mutableListOf<Merchant>()
-            var counter = 0
-            exploreRepository.getMerchantData(merchantDataSizeDB).collect {
-                counter++
-                merchantCache.add(it)
-                if (merchantCache.size == PAGE_SIZE) {
+
+            if (merchantDataSizeDB < merchantDataSize) {
+
+                if (merchantDataSizeDB > 0) {
+                    log.info("MERCHANT data partially synced ($merchantDataSizeDB of $merchantDataSize), continuing")
+                } else {
+                    log.info("MERCHANT data empty, fresh sync")
+                }
+                val merchantCache = mutableListOf<Merchant>()
+                var counter = 0
+                exploreRepository.getMerchantData(merchantDataSizeDB).collect {
+                    counter++
+                    merchantCache.add(it)
+                    if (merchantCache.size == PAGE_SIZE) {
+                        merchantDao.save(merchantCache)
+                        merchantCache.clear()
+                        log.info("MERCHANT $counter of ${merchantDataSize - merchantDataSizeDB} sync took $tableSyncWatch")
+                    }
+                }
+                if (merchantCache.size > 0) {
                     merchantDao.save(merchantCache)
                     merchantCache.clear()
-                    log.info("MERCHANT $counter of ${merchantDataSize - merchantDataSizeDB} sync took $tableSyncWatch")
                 }
+                tableSyncWatch.stop()
+                log.info("MERCHANT sync took $tableSyncWatch ($counter records)")
             }
-            if (merchantCache.size > 0) {
-                merchantDao.save(merchantCache)
-                merchantCache.clear()
-            }
-            tableSyncWatch.stop()
-            log.info("MERCHANT sync took $tableSyncWatch ($counter records)")
 
-            preferences.edit()
-                .putLong(PREFS_LAST_SYNC_KEY, Calendar.getInstance().timeInMillis)
-                .apply()
+            preferences.edit().apply {
+                putBoolean(PREFS_SYNC_IN_PROGRESS_KEY, false)
+                putLong(PREFS_LAST_SYNC_KEY, lastDataUpdate)
+            }.apply()
 
             log.info("Sync Explore Dash finished")
 
             return@withContext Result.success()
 
         } catch (ex: Exception) {
-            analytics.logError(ex, "syncing from $merchantDataSizeDB ($lastSync/$lastDataUpdate")
-            log.info("Sync Explore Dash not fully finished: ${ex.message}")
+            analytics.logError(ex, "syncing from $atmDataSizeDB, $merchantDataSizeDB ($lastSync/$lastDataUpdate")
```

Comment:
Closing parenthesis is missing

Reply:
Fixed missing ")".
The 1643285943457.dat is encrypted